ui/ui_vfinal/scripts/process_query.js

```diff
@@ -0,0 +1,269 @@
+ * @param {string} slices[].sliceOp - The name of slice operation applied 
+ * @param {(number|string|boolean|Array)} slices[].sliceVal - The slice values for filtering
+ *
+ * @param {?Object} dateRange - The date range in which results are to be fetched
+ * @param {string} dateRange.dateCol - The date column containing dates used for filtering results
+ * @param {string} dateRange.dateStart - The start date (yyyy-mm-dd)
+ * @param {string} dateRange.dateEnd - The end date (yyyy-mm-dd) 
+ *
+ * @param {?string} timeGranularity - The time granularity value selected by the user
+ *
+ * @param {?Object} comparisonValue - Values to be compared for slice-compare intent
+ * @paran {string} comparisonValue.comparisonColumn - The comparison column which contains values to comapre
+ * @param {string} comparisonValue.slice1 - The first value to be compared 
+ * @param {string} comparisonValue.slice2 - The second value to be compared (can be *)
+ *
+ * @param {?Object} compareDateRange - Date ranges to be compared for time-compare intent
+ * @paran {string} compareDateRange.dateCol - The date column which contains date ranges to comapre
+ * @param {string} compareDateRange.dateStart1 - The start date of first date-range
+ * @param {string} compareDateRange.dateEnd1 - The end date of first date-range
+ * @param {string} compareDateRange.dateStart2 - The start date of second date-range
+ * @param {string} compareDateRange.dateEnd2- The end date of second date-range
+ *
+ * @param {?Object} correlationMetrics - The metrics for correlation intent
+ * @param {string} correlationMetrics.metric1 - The first metric used to apply correlation
+ * @param {string} correlationMetrics.metric2 - The second metric used to apply correlation
+ *
+ * @param {Object} dateColumnsList - The object containing information about the date columns
+ * @param {string} dateColumnsList.dateColumnName.type - The type of date column 
+ * @param {boolean} dateColumnsList.dateColumnName.day_first - The format of date column
+ * @param {string} dateColumnsList.dateColumnName.min_date - The minimum date in specified date column
+ * @param {string} dateColumnsList.dateColumnName.max_date - The maximum date in specified date column
+ *
+ * @return {Object} queryResult - The results of the query submitted 
+ * @return {string} queryResult.status - Indicates whether gcp function returned results successfully
+ * @return {string} queryResult.jsonQuery - The json for the submitted query
+ * @return {Array} queryResult.outputTable - The output table of the query submitted 
+ * @return {Object[]} queryResult.suggestions - The list of suggestions related to the submitted query
+ * @return {?Array} queryResult.filtersPassed - The list of true-false values to insert output as column on filtering
+ * @return {?Array} queryResult.inTopK -  The list of true-false values to insert output as column in topk
+ */
+function evalQuery(
+    intent, 
+    metric, 
```

Comment:
if i understand it right , this is just intent name and aspects, if an intent doesnt have few aspects they will be null right ? if yes can you add this to the comment at the top.

Reply:
Also , can you link what are the aspects required for each intent information in the readme.

Reply:
I am using this notation: {?Array} to indicate null or array
Resource: https://jsdoc.app/tags-type.html
A number or null.
{?number} | This indicates that the type is either the specified type, or null.
-- | --

Reply:
Yeah cool sounds good.